core/kernel.go

```diff
@@ -73,300 +65,305 @@ const (
-			if err != nil {
-				return nil, err
-			}
-		}
+	kern.Logger.InfoMsg("State loading successful")
+	return nil
+}
 
-		err = kern.State.InitialCommit()
-		if err != nil {
-			return nil, err
-		}
+// LoadDump restores chain state from the given dump file
+func (kern *Kernel) LoadDump(genesisDoc *genesis.GenesisDoc, dbDir, restore string) (err error) {
+	stateDB, err := kern.LoadBlockchain(genesisDoc, dbDir)
 
-		if len(genesisDoc.AppHash) != 0 {
-			if !bytes.Equal(genesisDoc.AppHash, kern.State.Hash()) {
-				return nil, fmt.Errorf("AppHash does not match, got AppHash 0x%X expected 0x%s. Is the correct --restore-dump specified?", kern.State.Hash(), genesisDoc.AppHash.String())
-			}
-		}
+	kern.State, err = state.MakeGenesisState(stateDB, genesisDoc)
+	if err != nil {
+		return fmt.Errorf("could not build genesis state: %v", err)
 	}
 
-	kern.Logger.InfoMsg("State loading successful")
+	if len(kern.Blockchain.GenesisDoc().AppHash) == 0 {
+		return fmt.Errorf("AppHash is required when restoring chain")
+	}
+
+	reader, err := state.NewFileDumpReader(restore)
+	if err != nil {
+		return err
+	}
+
+	if err = kern.State.LoadDump(reader); err != nil {
+		return err
+	}
+
+	if err = kern.State.InitialCommit(); err != nil {
+		return err
+	}
 
-	txCodec := txs.NewAminoCodec()
-	tmGenesisDoc := tendermint.DeriveGenesisDoc(genesisDoc)
 	params := execution.ParamsFromGenesis(genesisDoc)
-	checker := execution.NewBatchChecker(kern.State, params, kern.Blockchain, kern.Logger)
+	kern.exeChecker = execution.NewBatchChecker(kern.State, params, kern.Blockchain, kern.Logger)
+	kern.exeCommitter = execution.NewBatchCommitter(kern.State, params, kern.Blockchain, kern.Emitter, kern.Logger, kern.exeOptions...)
 
-	kern.Emitter = event.NewEmitter(kern.Logger)
-	committer := execution.NewBatchCommitter(kern.State, params, kern.Blockchain, kern.Emitter, kern.Logger, exeOptions...)
+	kern.Logger.InfoMsg("State loading successful")
+	return nil
+}
+
+// GetNodeView builds and returns a wrapper of our tendermint node
+func (kern *Kernel) GetNodeView() (nodeView *tendermint.NodeView, err error) {
+	nodeView, err = tendermint.NewNodeView(kern.Node, kern.txCodec, kern.RunID)
+	return nodeView, err
+}
+
+// AddExecutionOptions extends our execution options
+func (kern *Kernel) AddExecutionOptions(opts ...execution.ExecutionOption) {
+	kern.exeOptions = append(kern.exeOptions, opts...)
+}
+
+// AddProcesses extends the services that we launch at boot
+func (kern *Kernel) AddProcesses(pl ...process.Launcher) {
+	kern.Launchers = append(kern.Launchers, pl...)
+}
+
+func (kern *Kernel) SetKeyClient(client keys.KeyClient) {
+	kern.keyClient = client
+}
+
+func (kern *Kernel) SetKeyStore(store *keys.KeyStore) {
+	kern.keyStore = store
+}
+
+// TendermintFromConfig loads our consensus engine into the kernel
+func (kern *Kernel) TendermintFromConfig(tmConf *tendermint.BurrowTendermintConfig, privVal tmTypes.PrivValidator, validator *crypto.Address) (err error) {
```

Comment:
I think this should be called `LoadTendermint` since it does not return `Tendermint` and there is no way to load without config.

Reply:
okay actually `LoadTendermintFromConfig` is good